packages/core/src/routes/interaction/index.test.ts

```diff
@@ -89,171 +102,191 @@ mockEsmDefault(
-      expect(submitInteraction).not.toBeCalled();
+      const response = await sessionRequest.put(`${interactionPrefix}/event`).send(body);
+      expect(verifySignInModeSettings).toBeCalled();
+      expect(storeInteractionResult).not.toBeCalled();
       expect(response.status).toEqual(404);
     });
+  });
 
-    it('Forgot event with forgot password event interaction session in record should call methods properly', async () => {
-      getInteractionStorage.mockResolvedValueOnce({ event: Event.ForgotPassword });
+  describe('PATCH /interaction/identifiers', () => {
+    const path = `${interactionPrefix}/identifiers`;
 
+    it('should update identifiers properly', async () => {
       const body = {
-        event: Event.ForgotPassword,
+        email: 'email@logto.io',
+        passcode: 'passcode',
       };
-
       const response = await sessionRequest.patch(path).send(body);
-      expect(verifyIdentifier).toBeCalled();
-      expect(verifyProfile).toBeCalled();
-      expect(validateMandatoryUserProfile).not.toBeCalled();
-      expect(submitInteraction).toBeCalled();
-      expect(response.status).toEqual(200);
-      expect(response.body).toEqual({ redirectUri: 'logto.io' });
+      expect(getInteractionStorage).toBeCalled();
+      expect(verifyIdentifierPayload).toBeCalled();
+      expect(mergeIdentifiers).toBeCalled();
+      expect(storeInteractionResult).toBeCalled();
+      // Supertest does not return the error body
+      expect(response.status).toEqual(204);
     });
+  });
 
-    it('Forgot event with sign-in event interaction session in record should call methods properly', async () => {
-      getInteractionStorage.mockResolvedValueOnce({ event: Event.SignIn });
+  describe('/interaction/profile', () => {
+    const path = `${interactionPrefix}/profile`;
+    const sessionRequest = createRequester({
+      anonymousRoutes: interactionRoutes,
+      provider: createMockProvider(jest.fn().mockResolvedValue(baseProviderMock)),
+    });
 
+    it('PATCH /interaction/profile', async () => {
       const body = {
-        event: Event.ForgotPassword,
+        email: 'email@logto.io',
       };
-
       const response = await sessionRequest.patch(path).send(body);
-      expect(verifyIdentifier).not.toBeCalled();
-      expect(verifyProfile).not.toBeCalled();
-      expect(validateMandatoryUserProfile).not.toBeCalled();
-      expect(submitInteraction).not.toBeCalled();
-      expect(response.status).toEqual(404);
+      expect(verifyProfileSettings).toBeCalled();
+      expect(getInteractionStorage).toBeCalled();
+      expect(storeInteractionResult).toBeCalled();
+      expect(response.status).toEqual(204);
+    });
+
+    it('DELETE /interaction/profile', async () => {
+      const response = await sessionRequest.delete(path);
+      expect(getInteractionStorage).toBeCalled();
+      expect(storeInteractionResult).toBeCalled();
+      expect(response.status).toEqual(204);
```

Comment:
When is this be called? I don't recall we have a scenario of deleting profile from interaction.

Reply:
register with an existing email then switches to sign-in.  Need to delete the email profile.

Reply:
ok